src/System.Runtime.Loader/tests/DefaultContext/DefaultLoadContextTest.cs

```diff
@@ -58,24 +58,43 @@ public static void LoadInDefaultContext()
         {
             Init();
 
-           // This will attempt to load an assembly, by path, in the Default Load context via the Resolving event
-           var assemblyName = "System.Runtime.Loader.Noop.Assembly";
+            // This will attempt to load an assembly, by path, in the Default Load context via the Resolving event
+            var assemblyNameStr = "System.Runtime.Loader.Noop.Assembly";
+            var assemblyName = new AssemblyName(assemblyNameStr);
 
+            // By default, the assembly should not be found in DefaultContext at all
+            Assert.Throws(typeof(FileNotFoundException), () => Assembly.Load(assemblyName));
+
+            // Now, wireup the Resolving event to locate the assembly
             AssemblyLoadContext.Default.Resolving += ResolveAssembly;
-            var assemblyExpected = AssemblyLoadContext.Default.LoadFromAssemblyName(new AssemblyName(assemblyName));
+            
+            // This will invoke the resolution via VM requiring to bind using the TPA binder
+            var assemblyExpectedFromLoad = Assembly.Load(assemblyName);
+
+            // We should have successfully loaded the assembly in default context.
+            Assert.NotNull(assemblyExpectedFromLoad);
+
+            // And make sure the simple name matches
+            Assert.Equal(assemblyNameStr, assemblyExpectedFromLoad.GetName().Name);
+
+            // This will resolve the assembly via event invocation.
+            var assemblyExpected = AssemblyLoadContext.Default.LoadFromAssemblyName(assemblyName);
             
             // We should have successfully loaded the assembly in default context.
             Assert.NotNull(assemblyExpected);
 
+            // What we got via Assembly.Load and LoadFromAssemblyName should be the same
+            Assert.Equal(assemblyExpected, assemblyExpectedFromLoad);
+
             // And make sure the simple name matches
-            Assert.Equal(assemblyExpected.GetName().Name, assemblyName);
+            Assert.Equal(assemblyExpected.GetName().Name, assemblyNameStr);
 
             // Unwire the Resolving event.
             AssemblyLoadContext.Default.Resolving -= ResolveAssembly;
```

Comment:
> No since this is not an invariant.

We iterate through the registered handlers in a specific order.  An application could easily take a dependency on that, and if we inadvertently changed the order, we wouldn't know until that customer's application broke.  Why is it not valuable to have a test that alerts us if we do accidentally change this?

Separately, what if we had a bug and we ended up not exiting out early on first success.  Wouldn't a test that validates that we only invoke up to the first successful result be useful?

And what if one of the handlers throws an exception.  What behavior would we expect then, and shouldn't we have a test for it, even if the expected behavior is crashing?